desertbot/modules/admin/Update.py

```diff
@@ -60,33 +65,68 @@ def execute(self, message: IRCMessage):
             finally:
                 response += " | No auto-reload due to requirements change, please restart bot."
         else:
+            changedPyFiles = [path for path in changedFiles if path.endswith(".py")]
+            addedPyFiles = [path for path in addedFiles if path.endswith(".py")]
+
             modulesToReload = []
-            for filepath in changedFiles:
-                filename = filepath.split(os.path.sep)[-1]  # changedFiles contains full filepaths, split on os.path.sep and get last for filename
-                if filename in self.bot.moduleHandler.fileMap:
-                    modulesToReload.append(self.bot.moduleHandler.fileMap[filename])
+            modulesToLoad = []
+
+            for filepath in changedPyFiles:
+                filename = filepath.split(os.path.sep)[-1]  # list contains full filepaths, split on os.path.sep and get last for filename
+                if "modules" in filepath:
+                    if filename in self.bot.moduleHandler.fileMap:
+                        modulesToReload.append(self.bot.moduleHandler.fileMap[filename])
                 else:
                     modulesToReload = []
+                    modulesToLoad = []
                     response += " | No auto-reload due to change(s) in bot core, please restart bot."
                     self.logger.info("No auto-reload due to change in file {!r}".format(filename))
                     break
 
+            for filepath in addedPyFiles:
+                filename = filepath.split(os.path.sep)[-1]  # list contains full filepaths, split on os.path.sep and get last for filename
+                if "modules" in filepath:
+                    if filename in self.bot.moduleHandler.fileMap:
```

Comment:
I don't think newly added files will be in the fileMap?
Otherwise this all looks great.